app/src/androidTest/java/ch/epfl/qedit/answer/MatrixFragmentTest.java

```diff
@@ -82,22 +100,77 @@ public void type(String input, String expected) {
 
     @Test
     public void testCanEnterNumbersInFields() {
+        init();
         type("1232", "1232");
     }
 
     @Test
     public void testCanOnlyHaveOneMinusSign() {
+        init();
         type("--12", "-12");
     }
 
     @Test
     public void testCanOnlyHaveOneDecimalPoint() {
+        init();
         type("23..2", "23.2");
     }
 
     @Test
     public void testCantEnterMoreDigitsThanMaxCharacters() {
-        // MaxCharacters = 5 for MatrixFormat.createMatrix3x3()
+        // MaxCharacters = 5 by default
+        init();
         type("123456", "12345");
     }
+
+    @Test
+    public void testAnswerIsSavedInQuizViewModel() {
+        init();
+        quizViewModel.getFocusedQuestion().postValue(0);
+        assertNull(quizViewModel.getAnswers().getValue().get(0));
+
+        int id = testRule.getFragment().getId(0, 0);
+        onView(withId(id)).perform(click());
+        onView(withId(id)).perform((typeText("1232"))).perform(closeSoftKeyboard());
+
+        assertNotNull(quizViewModel.getAnswers().getValue().get(0));
+        assertEquals(
+                "1232",
+                ((MatrixModel) quizViewModel.getAnswers().getValue().get(0)).getAnswer(0, 0));
+    }
+
+    // @Test TODO
```

Comment:
you have a TODO here, in case you forgot :)

Reply:
This is one of the two tests I still need to do.